src/session/client-certificate.c

```diff
@@ -102,6 +117,40 @@ read_proc_self_cgroup (size_t *out_length)
   return result;
 }
 
+static
+unsigned long long get_proc_pid_start_time (int dirfd)
+{
+  int fd = openat (dirfd, "stat", O_RDONLY);
+  if (fd < 0)
+    err (EX, "Failed to open /proc/pid/stat");
+
+  char buffer[4096];
+  ssize_t size = read (fd, buffer, sizeof buffer);
+  close (fd);
+  if (size < 0)
+    err (EX, "Failed to read /proc/pid/stat");
```

Comment:
You shouldn't call `close()` before calling `err()` — you might get the wrong `errno` reported.  Just move this above the `close()` I'd say.

It might also be worth checking for short reads (can we get interrupted reads on procfs?  Probably not...) and the case where we read exactly the requested number of bytes (because it probably means there is more).

You could also check to make sure we read a newline at the end.  Or: use stdio.

Reply:
But we don't know the exact number of bytes in advance. The idea is that as long as there are at least 22 items in the space-separated list (which the code enforces), it's uninteresting what follows after. The kernel may append new fields in the future, but the existing fields are API.

Reply:
Fixed the close()